modules/tinymce/src/plugins/accordion/main/ts/core/Actions.ts

```diff
@@ -0,0 +1,39 @@
+import { Id } from '@ephox/katamari';
+
+import Editor from 'tinymce/core/api/Editor';
+
+import * as Events from '../api/Events';
+
+const insertAccordion = (editor: Editor): void => {
+  const container = editor.selection.getNode();
+  if (container.nodeName === 'SUMMARY') {
+    const body = editor.dom.getNext(container, 'div.mce-accordion-body');
+    if (!body) {
+      return;
+    }
+    if (!body.lastChild) {
+      return;
+    }
+    editor.selection.setCursorLocation(body.lastChild, 1);
+  }
+
+  const uid = Id.generate('acc');
+  const summaryText = editor.dom.encode(editor.selection.getRng().toString() || editor.translate('Accordion summary...'));
+  const bodyText = editor.dom.encode(editor.translate('Accordion body...'));
+  editor.insertContent(`<details data-mce-id="${uid}" class="mce-accordion" open="open"><summary class="mce-accordion-summary">${summaryText}</summary><div class="mce-accordion-body"><p>${bodyText}</p></div></details>`);
+
+  const details = editor.dom.select(`[data-mce-id="${uid}"]`)[0];
+  if (!details) {
+    return;
+  }
+  details.removeAttribute('data-mce-id');
+
+  const summary = editor.dom.select(`summary`, details)[0];
+  if (summary) {
+    editor.selection.setCursorLocation(summary, 1);
+  }
+
+  Events.fireInsertAccordionEvent(editor, details);
```

Comment:
That's not quite how TinyMCE events work. If the event name is present tense, like "InsertAccordion", it should give the integrator a chance to adjust what happens. A notification event must be past tense, e.g. "InsertedAccordion".

For example, the `insertCustomChar` event inserts the character returned by the event:
https://github.com/tinymce/tinymce/blob/c12fb6af3b6b72886946dd3fb1bb1b8be249949f/modules/tinymce/src/plugins/charmap/main/ts/core/Actions.ts#L5-L8

I think, given our goals for this feature, it should remain a notification event and be renamed.

- [ ] rename event (I realise this will require discussion, as you've followed the spec exactly)

Reply:
It is a good point, thanks! Will rename the event and will trigger the discussion of the necessity of the `InsertAccordion` event to manipulate its contents before inserting or something.